api/v0/metadata.go

```diff
@@ -19,55 +22,183 @@ func (e ErrInvalidMetadata) Error() string {
 
 // Metadata is data that provides information about retrieving
 // data for an index, from a particular provider.
-type Metadata struct {
-	// ProtocolID defines the protocol used for data retrieval.
-	ProtocolID multicodec.Code
-	// Data is specific to the identified protocol, and provides data, or a
-	// link to data, necessary for retrieval.
-	Data []byte
+type Metadata []byte
```

Comment:
that's true. and i'm not totally sold on this, but we should have an answer we're happy with for how to how to handle providers that speak multiple protocols.

* do we stick to our initial plan of different records for each protocol? what does the ingest interface look like to make it easier for providers to advertise multiple protocols?
* do we change the ingest format as well, putting a payload length after each payload, so we can parse the list even when there are protocol ids that aren't understood?

Reply:
As long as the outer Metadata is:
```go
type Metadata struct {
	ProtocolID multicodec.Code
	Data []byte
}
```
Then anything can go inside the Data, and we can find an efficient way to pack it and delimit it.  It could be a list of protocol IDs with a single metadata, or a list of separate complete Metadata values, and the outer encoding tells what to expect inside.